ocaml/xapi/xapi_pool.ml

```diff
@@ -3406,21 +3406,54 @@ let enable_tls_verification ~__context =
   | Some self ->
       Xapi_cluster_host.set_tls_config ~__context ~self ~verify:true
 
-let contains_bundle_repo ~__context ~repos =
-  List.exists
-    (fun repo -> Db.Repository.get_origin ~__context ~self:repo = `bundle)
+let assert_single_repo_can_be_enabled ~__context ~repos =
+  let origins =
     repos
+    |> List.filter_map (fun repo ->
+           match Db.Repository.get_origin ~__context ~self:repo with
+           | (`bundle | `remote_pool) as origin ->
+               Some origin
+           | `remote ->
+               None
+       )
+    |> List.fold_left
+         (fun acc origin -> if List.mem origin acc then acc else origin :: acc)
+         []
+  in
+  match (repos, origins) with
+  | _ :: _ :: _, _ :: _ ->
```

Comment:
only
`match origins with | _::_`
is enough?

Reply:
Yes, it's enough. We just check the count of origins is larger than 0.